benchmark/arrayops/arrayops.d

```diff
@@ -0,0 +1,103 @@
+/**
+ * Benchmark for array ops.
+ *
+ * Copyright: Copyright Martin Nowak 2016 -.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:    Martin Nowak
+ */
+import core.cpuid, std.algorithm, std.datetime, std.meta, std.stdio, std.string,
+    std.range;
+
+float[6] getLatencies(T, string op)()
+{
+    enum N = 256;
+    auto a = new T[](N), b = new T[](N), c = new T[](N);
+    a[] = 3;
+    b[] = 2;
+    c[] = 1;
+    float[6] latencies = float.max;
+    foreach (i, ref latency; latencies)
+    {
+        auto len = 1 << i;
+        foreach (_; 1 .. 32)
+        {
+            auto sw = StopWatch(AutoStart.yes);
+            foreach (off; 0 .. 1_024)
+            {
+                off &= 127;
+                enum op = op.replace("const", "1").replace("a",
```

Comment:
Maybe '1' isn't the best generic constant, as there might be special optimizations for `mul` or `div`. I'm not sure how to avoid overflow and underflow with other constants for all types, though.